phantazm-zombies/src/main/java/com/github/phantazmnetwork/zombies/equipment/gun/shoot/ShootTester.java

```diff
@@ -3,13 +3,38 @@
 import com.github.phantazmnetwork.zombies.equipment.gun.GunState;
 import org.jetbrains.annotations.NotNull;
 
+/**
+ * Tests for a gun's ability to shoot.
+ */
 public interface ShootTester {
+
+    /**
+     * Tests if the gun should shoot.
+     * A gun may be able to fire as determined by {@link #canFire(GunState)} while not being able to shoot.
```

Comment:
what is the difference between "fire" and "shoot"

Reply:
ultimate pistols for example fire twice every single time they shoot
it's a little difficult to express the difference

Reply:
okay that makes sense, but i think you should
a) change the wording a bit (don't really know what to)
b) elaborate the difference in the javadoc more (clarifying there can be multiple fires per shot)

Reply:
a is a little hard I'll do b